baggage/baggage.go

```diff
@@ -304,9 +304,33 @@ func parseMember(member string) (Member, error) {
 	if err != nil {
 		return newInvalidMember(), fmt.Errorf("%w: %w", errInvalidValue, err)
 	}
+
+	if !utf8.ValidString(value) {
+		// W3C baggage spec:
+		// https://github.com/w3c/baggage/blob/main/baggage/HTTP_HEADER_FORMAT.md?plain=1#L69
+		value = replaceInvalidUTF8Sequences(value)
+	}
+
 	return Member{key: key, value: value, properties: props, hasData: true}, nil
 }
 
+func replaceInvalidUTF8Sequences(input string) string {
+	var output []rune
+	for i := 0; i < len(input); {
+		r, size := utf8.DecodeRuneInString(input[i:])
+
+		if r == utf8.RuneError && size == 1 {
+			// Invalid UTF-8 sequence found, replace it with '�'
+			output = append(output, '�')
+		} else {
+			output = append(output, r)
+		}
+		i += size
+	}
+
+	return string(output)
```

Comment:
This implementation is very inefficient. It does a lot of unnecessary heap allocations which can be on a hot path if someone uses invalid UTF-8 sequences. Please take a look at `strings.Replace` implementation for inspiration. https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/strings/strings.go;l=1060

Reply:
Done